lmfit/confidence.py

```diff
@@ -190,8 +182,10 @@ def __init__(self, minimizer, result, p_names=None, prob_func=None,
         if nvars < 2:
             raise MinimizerException(CONF_ERR_NVARS)
 
-        if prob_func is None or not hasattr(prob_func, '__call__'):
+        if prob_func is None:
             self.prob_func = f_compare
+        else:
+            self.prob_func = prob_func
```

Comment:
I disagree, we don't check the correctness elsewhere, why do it here for that quite seldom used parameter? Especially since before the failure was silent which is a completely unexpected behaviour. Also if someone provides a non callable input he will get a very informative stackstrace.

Reply:
we certainly check if functions are ```callable``` elsewhere. But I do see your point about a having a silent failure previously when providing a non-callable. I'd argue though that we should improve the type-checking and error-handling instead of removing the check.

Reply:
I still disagree, we don't these kind of checks elsewhere, why start here for that obscure parameter which obviously nobody uses expect me (else somebody would have likley reported that bug). Again, if you supply something wrong, you will get quite an informative error trance.